pkg/prove/querier.go

```diff
@@ -0,0 +1,56 @@
+package prove
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	"github.com/tendermint/tendermint/types"
+)
+
+const TxInclusionQueryPath = "txInclusionProof"
+
+// Querier defines the logic performed when the ABCI client using the Query
+// method with the custom prove.QueryPath. The index of the transaction being
+// proved must be appended to the path. The marshalled bytes of the transction
+// proof (tmproto.TxProof) are returned.
+//
+// example path for proving the third transaction in that block:
+// custom/txInclusionProof/3
+func QueryTxInclusionProof(_ sdk.Context, path []string, req abci.RequestQuery) ([]byte, error) {
+	// parse the index from the path
+	if len(path) != 1 {
+		return nil, fmt.Errorf("unexpected query path length %d", len(path))
```

Comment:
[Non Blocking][Cosmetics]
```suggestion
		return nil, fmt.Errorf("expected query path length: 1 actual: %d ", len(path))
```